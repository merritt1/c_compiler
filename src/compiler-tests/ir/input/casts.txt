void main(void) {
	int i, j;
	i = (int)(char)j;
}