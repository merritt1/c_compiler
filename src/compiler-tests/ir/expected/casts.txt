(procBegin,  main)
(addressOf,     r0,      i)
(addressOf,     r1,      j)
(loadWord,     r2,      r1)
(castWordToByte,     r3,      r2)
(castSignedByteToWord,     r4,      r3)
(storeWord,     r0,      r4)
(procEnd ,  main)
